kong/db/dao/init.lua

```diff
@@ -145,6 +146,24 @@ local function validate_options_value(options, schema, context)
     errors.ttl = fmt("cannot be used with '%s'", schema.name)
   end
 
+  if schema.tags == true and options.tags ~= nil then
+    if context ~= "select" then
+      errors.tags = fmt("option can only be used with selects and pages, not with '%ss'",
+                       tostring(context))
```

Comment:
> 2\. (I hope we don't need to support `/` in tag names in the future!)

I think the trick is to:
1. split non url-decoded
2. then url-decode each tag

which means you can have `/` (in a future) if you url encode them.

Reply:
> magic `tags = true` attribute

I think it is that we wanted that to be `option`, but it is now an implicit field (or column). Also the original PG one didn't have it as a field. But well, now both C* and PG it is a field, so perhaps that means that it should be field in schemas too. Though a lot of automation will be harder to do. As we cannot check if entity schema is taggable. We can check if it has `tags` field, but we cannot be sure if it was meant for tagging or just a field. Now it is quite easy to check `if schema.tags then -- create triggers, add tags field etc. `.

One option is to have both, then we don't need that implicit field.

Reply:
> But well, now both C* and PG it is a field, so perhaps that means that it should be field in schemas too. 

Yes, that was my reasoning.

> We can check if it has tags field, but we cannot be sure if it was meant for tagging or just a field.

We could add a small validation for that in `kong.db.schema.entity` to make sure that any `tags` field has the correct "type signature".

Reply:
If I read the conversation here correctly, is the path forward with the `tags=true` attribute  having both the attribute and the field definition in schema?

Reply:
Just a regular field definition in the schema, no attribute. Then instead of checking for tags with `if schema.tags`, do `if schema.fields.tags`, should work just as well.

If we were concerned that the `tags` fields in entities is indeed of the correct type, we could add a check for that in `Entity.new` in `kong/db/schema/entity.lua`. But I don't really think that's necessary. Just using `{ tags = typedefs.tags }` in schema definitions should be enough, like we do for `id`, `created_at`, etc.